dashboard/app/models/script.rb

```diff
@@ -114,6 +114,17 @@ def hide_pilot_scripts
 
   validate :set_is_migrated_only_for_migrated_scripts
 
+  def prevent_duplicate_levels
+    reload
+
+    unless levels.count == levels.uniq.count
+      h = {}
+      levels.map(&:key).each {|key| h[key] = (h[key] || 0) + 1}
```

Comment:
I think you could use group_by to simplify this logic slightly, like `duplicate_keys = levels.map(&:key).group_by {|key| key}.select {|_key, values| values.count > 1}.keys`

But I don't think it makes much of a difference so up to you.

Reply:
💡 ! I have looked up how to do "count by" in ruby many times and never thought to use `group_by`. Thanks for this!